iocage/lib/Jail.py

```diff
@@ -0,0 +1,575 @@
+        defaultrouter6 = self.config.defaultrouter6
+
+        if not defaultrouter or defaultrouter6:
+            self.logger.spam("no static routes configured")
+            return
+
+        if defaultrouter:
+            self.logger.verbose(
+                f"setting default IPv4 gateway to {defaultrouter}",
+                jail=self
+            )
+            self._set_route(defaultrouter)
+
+        if defaultrouter6:
+            self._set_route(defaultrouter6, ipv6=True)
+
+    def _set_route(self, gateway, ipv6=False):
+
+        ip_version = 4 + 2 * (ipv6 is True)
+
+        self.logger.verbose(
+            f"setting default IPv{ip_version} gateway to {gateway}",
+            jail=self
+        )
+
+        command = [
+            "/sbin/route",
+            "add"
+        ] + (["-6"] if (ipv6 is True) else []) + [
+            "default",
+            gateway
+        ]
+
+        self.exec(command)
+
+    def require_jail_not_existing(self):
+        if self.exists:
+            msg = f"Jail {self.humanreadable_name} already exists"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+    def require_jail_existing(self):
+        if not self.exists:
+            msg = f"Jail {self.humanreadable_name} does not exist"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+    def require_jail_stopped(self):
+        if self.running:
+            msg = f"Jail {self.humanreadable_name} is already running"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+    def require_jail_running(self):
+        if not self.running:
+            msg = f"Jail {self.humanreadable_name} is not running"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+    def update_jail_state(self):
+        try:
+            stdout = subprocess.check_output([
+                "/usr/sbin/jls",
+                "-j",
+                self.identifier,
+                "-v",
+                "-h"
+            ], shell=False, stderr=subprocess.DEVNULL)
+            output = stdout.decode("utf-8").strip()
+
+            keys, values = [x.split(" ") for x in output.split("\n")]
+            self.jail_state = dict(zip(keys, values))
+
+        except:
+            self.jail_state = None
+
+    def _teardown_mounts(self):
+
+        mountpoints = list(map(
+            lambda mountpoint: f"{self.path}/root{mountpoint}",
+            [
+                "/dev/fd",
+                "/dev",
+                "/proc",
+                "/root/compat/linux/proc"
+            ]
+        ))
+
+        mountpoints += list(map(lambda x: x["destination"],
+                                list(self.config.fstab)))
+
+        for mountpoint in mountpoints:
+            if os.path.isdir(mountpoint):
+                iocage.lib.helpers.umount(
+                    mountpoint,
+                    force=True,
+                    ignore_error=True  # maybe it was not mounted
```

Comment:
We do not check in any place if a mountpoint is mounted or not. Unless we use a custom mount/umount lib I tend to leave it as it is, because later on the umount function should not attempt to unmount some directory that is not mounted. I'm planning to implement a FreeBSD mount/umount lib in the near future, so this will most probably not in here for long.